<launch>

	<!-- Start Simulator -->
	<!-- include the control part -->
	<include file="$(find course_agv_gazebo)/launch/course_agv_world.launch"/>
    <!-- Publish global pose -->
    <node pkg="course_agv_gazebo" type="robot_tf.py" output="screen" name="robot_tf"/>
	
	<!-- Run the map server -->
	<arg name="map_file" default="$(find course_agv_gazebo)/config/map.yaml"/>
	<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)"/>

	<node pkg="course_agv_slam_task" type="particle_filter" name="particle_filter" output="screen">
		
		<!--from a rosbag-->
        <param name="init_x" value="-6.0"/>
        <param name="init_y" value="6.0"/>
        <param name="init_theta" value="0.0"/>

        <param name="init_rand_xy" value="0.5"/>
        <param name="init_rand_theta" value="0.3"/>

        <param name="std_map" value="0.1"/>
        <param name="propag_range" value="3"/>     

        <param name="new_ratio" value="0.1"/>
        <!-- <param name="N_thres" value="8900"/>    -->
        <param name="N_thres" value="9500"/>   

	</node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find course_agv_slam_task)/rviz/one_for_all.rviz" />
</launch>
